actions/diagnostics_actions.jsx

```diff
@@ -77,7 +84,37 @@ export function measure(name1, name2) {
     return [lastDuration, measurementName];
 }
 
+export function trackLoadTime() {
+    if (!isDevMode() || !isSupported([performance.timing.loadEventEnd, performance.timing.navigationStart])) {
+        return;
+    }
+
+    // Must be wrapped in setTimeout because loadEventEnd property is 0
+    // until onload is complete, also time added because analytics
+    // code isn't loaded until a subsequent window event has fired.
+    const tenSeconds = 10000;
+    setTimeout(() => {
+        const {loadEventEnd, navigationStart} = window.performance.timing;
+        const pageLoadTime = loadEventEnd - navigationStart;
+        trackEvent('performance', 'page_load', {duration: pageLoadTime});
+    }, tenSeconds);
+}
+
 function mostRecentDurationByEntryName(entryName) {
     const entriesWithName = performance.getEntriesByName(entryName);
     return entriesWithName.map((item) => item.duration)[entriesWithName.length - 1];
+}
+
+function isSupported(checks) {
```

Comment:
Maybe we could just do this once and store the results? That might help reduce the impact of these checks

Reply:
In redux?

Reply:
Done.

Reply:
Done.